### PR TITLE
WIP-164 develop 3d classification inference example

### LINKED ISSUE #164: develop 3d classification inference example
**Is your feature request related to a problem? Please describe.**
Now we already have array/dict based training/inference examples of 3D segmentation.
But only have 1 array based training example of 3D classification.
Need to develop:
(1) dict based 3D classification training example.
(2) array based 3D classification inference example.
(3) dict based 3D classification inference example.
(4) [optional] save classification results into CSV file.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #164 .

### Description
This PR implemented:
1. 3D classification training based on dictionary format data and transforms.
2. 3D classification inference based on dictionary format data and transforms.
3. 3D classification inference based on array format data and transforms.
4. developed ClassificationSaver event handler.
5. added Resized, Rescaled transforms.
6. fixed bug in Resize transform that resize by channel dim an updated unit test.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/handlers/classification_saver.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# limitations under the License.
+
+import os
+import csv
+import numpy as np
+import torch
+from ignite.engine import Events
+
+
+class ClassificationSaver:
```
Comment (wyli) [inline monai/handlers/classification_saver.py]: could you provide a unit test?

File: monai/handlers/classification_saver.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        """
+        meta_data = self.batch_transform(engine.state.batch)
+        filenames = meta_data['filename_or_obj']
+
+        engine_output = self.output_transform(engine.state.output)
+        for batch_id, filename in enumerate(filenames):  # save a batch of files
+            output = engine_output[batch_id]
+            if isinstance(output, torch.Tensor):
+                output = output.detach().cpu().numpy()
+            self._prediction_dict[filename] = output
```
Comment (wyli) [inline monai/handlers/classification_saver.py]: this potentially creates inconsistency with respect to line 48 of this file (which has data type conversion)?

File: monai/handlers/classification_saver.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    def finalize(self):
+        """
+        Writes the prediction dict to a csv
+
+        """
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+        with open(self._preds_filepath, 'w') as f:
+            for k, v in sorted(self._prediction_dict.items()):
```
Comment (wyli) [inline monai/handlers/classification_saver.py]: would this `sort` mess up the order of existing csv files we load in the constructor?

File: monai/handlers/classification_saver.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                batch dimension. Each item in the batch will be saved individually.
+            name (str): identifier of logging.logger to use, defaulting to `engine.logger`.
+
+        """
+        self.output_dir = output_dir
+        self._prediction_dict = {}
+        self._preds_filepath = os.path.join(output_dir, 'predictions.csv')
+        if not overwrite:
+            if os.path.exists(self._preds_filepath):
+                with open(self._path, 'r') as f:
```
Comment (wyli) [inline monai/handlers/classification_saver.py]: perhaps we can do this reading right before we write final new results?
